File: app/src/main/java/com/wmbest/hue/MainActivity.java
package com.wmbest.hue;

import android.app.Activity;

public class MainActivity extends Activity {

    final Channel mChannel;
    final WifiP2pManager mManager;
    WifiP2pServiceRequest mRequester;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mManager = (WifiP2pManager) getSystemService(Context.WIFI_P2P_SERVICE);
        mChannel = mManager.initialize(this, getMainLooper(), new ChannelListener() {

            public void onChannelDisconnected() {
                Log.i("CI", "Channel detected!");

            }
        });
        mManager.setUpnpServiceResponseListener(mChannel, new UpnpServiceResponseListener() {

            public void onUpnpServiceAvailable(List<String> arg0, WifiP2pDevice arg1) {
                Log.i("sd", "Found device!!");          
            }

        });
        mRequester = WifiP2pUpnpServiceRequest.newInstance();

        mManager.addServiceRequest(mChannel, mRequester, new ActionListener() {

            public void onSuccess() {

                Log.i("d", "AddServiceRequest success!");

                mManager.discoverServices(mChannel, new ActionListener() {

                    public void onSuccess() {
                        Log.i("d", "DiscoverServices success!");
                    }

                    public void onFailure(int reason) { 
                    }
                });

            }

            public void onFailure(int reason) {
            }
        });

    } 
} 

File: app/src/main/kotlin/com/wmbest/hue/MainActivity.kt
package com.wmbest.hue

import android.os.Bundle
import android.support.v7.app.ActionBarActivity

import kotlinx.android.synthetic.activity_main.*

import com.wmbest.hue.fragments.*
import com.wmbest.hue.util.*

class MainActivity : ActionBarActivity() {

    override fun onCreate(state: Bundle?) {
        super.onCreate(state)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        if (state == null) {
            replace(R.id.container, BridgeSetupFragment(), "setup").commit()
        }
    } 
} 
